optimum/onnxruntime/quantization.py

```diff
@@ -0,0 +1,262 @@
+                per_channel=self.ort_config.per_channel,
+                reduce_range=self.ort_config.reduce_range,
+                activation_type=self.activation_type,
+                weight_type=self.weight_type,
+                optimize_model=self.ort_config.optimize_model,
+                use_external_data_format=self.ort_config.use_external_data_format,
+                calibrate_method=self.calibrate_method,
+            )
+        else:
+            raise ValueError(
+                f"Unknown quantization approach: `quantization_approach` was set to {self.quantization_approach}. "
+                f"Supported quantization approaches are " + ", ".join(SUPPORTED_QUANT_MODE)
+            )
+
+    def get_calib_dataset(self) -> Dataset:
+        """
+        Returns the calibration :class:`~datasets.arrow_dataset.Dataset` to use for the post-training static
+        quantization calibration step.
+        """
+        if self.dataset_name is None:
+            raise ValueError(
+                "ORTQuantizer: Static quantization calibration step requires a dataset_name if no calib_dataset is "
+                "provided."
+            )
+        if self.preprocess_function is None:
+            raise ValueError(
+                "ORTQuantizer: Processing function to apply after loading the dataset used for static quantization "
+                "calibration step was not provided."
+            )
+        calib_dataset = load_dataset(
+            self.dataset_name,
+            name=self.dataset_config_name,
+            data_files=self.data_files,
+            split=self.ort_config.split,
+            cache_dir=self.cache_dir,
+        )
+        calib_dataset = calib_dataset.map(self.preprocess_function, batched=True)
+        return calib_dataset
+
+    def get_calib_dataloader(self, calib_dataset: Optional[Dataset] = None) -> DataLoader:
+        """
+        Returns the calibration :class:`~torch.utils.data.DataLoader`.
+        Args:
+            calib_dataset (:obj:`torch.utils.data.Dataset`, `optional`):
+                If provided, will override :obj:`self.calib_dataset`.
+        """
+        if calib_dataset is None and self.calib_dataset is None:
+            raise ValueError("ORTQuantizer: static quantization calibration step requires a calib_dataset.")
+
+        calib_dataset = calib_dataset if calib_dataset is not None else self.calib_dataset
+
+        if self.ort_config.max_samples is not None and len(calib_dataset) > self.ort_config.max_samples:
+            calib_dataset = calib_dataset.select(range(self.ort_config.max_samples))
+
+        ignored_columns = list(set(calib_dataset.column_names) - set(self.onnx_config.inputs.keys()))
+        calib_dataset = calib_dataset.remove_columns(ignored_columns)
+
+        generator = torch.Generator()
+        generator.manual_seed(int(torch.empty((), dtype=torch.int64).random_().item()))
+        sampler = RandomSampler(calib_dataset, generator=generator)
```

Comment:
Let's expose the seed as an optional parameter in the method, this would be expected by users to be able to reproduce quantization experiment.

Let's potentially also store the seed within the serialized config 👍🏻

Reply:
Good point !